veracruz-server/src/server.rs

```diff
@@ -16,80 +16,58 @@ use crate::platforms::icecap::VeracruzServerIceCap as VeracruzServerEnclave;
 use crate::platforms::linux::veracruz_server_linux::VeracruzServerLinux as VeracruzServerEnclave;
 #[cfg(feature = "nitro")]
 use crate::platforms::nitro::veracruz_server_nitro::VeracruzServerNitro as VeracruzServerEnclave;
-use bincode;
 use policy_utils::policy::Policy;
 use std::io::{Read, Write};
-use std::net::{Shutdown, TcpListener, TcpStream};
+use std::net::{TcpListener, TcpStream};
 use std::sync::{Arc, Mutex};
 use std::thread;
 
 type EnclaveHandlerServer = Box<dyn crate::common::VeracruzServer + Sync + Send>;
 type EnclaveHandler = Arc<Mutex<Option<EnclaveHandlerServer>>>;
 
-fn veracruz_server_request(
+fn handle_veracruz_server_request(
     enclave_handler: EnclaveHandler,
-    input_data: &[u8],
-) -> Result<Vec<u8>, VeracruzServerError> {
-    if input_data.len() < 4 {
-        return Err(VeracruzServerError::InvalidRequestFormatError);
-    }
-    let session_id = match bincode::deserialize(&input_data).unwrap() {
-        0 => {
-            let mut enclave_handler_locked = enclave_handler.lock()?;
-
-            let enclave = enclave_handler_locked
-                .as_mut()
-                .ok_or(VeracruzServerError::UninitializedEnclaveError)?;
-
-            enclave.new_tls_session()?
-        }
-        n @ 1u32..=std::u32::MAX => n,
-    };
-
-    let received_data = &input_data[4..];
-
-    let (active_flag, output_data_option) = {
+    mut stream: TcpStream,
+) -> Result<(), VeracruzServerError> {
+    let session_id = {
         let mut enclave_handler_locked = enclave_handler.lock()?;
-
         let enclave = enclave_handler_locked
             .as_mut()
             .ok_or(VeracruzServerError::UninitializedEnclaveError)?;
-
-        enclave.tls_data(session_id, received_data.to_vec())?
+        enclave.new_tls_session()?
     };
 
-    // Shutdown the enclave
-    if !active_flag {
-        let mut enclave_handler_locked = enclave_handler.lock()?;
+    loop {
+        // This buffer size gave close to optimal performance for
```

Comment:
Interesting! Have you tried different sizes? This should probably be moved to the top as a const

Reply:
I tried it with some different sizes for the buffer, but I didn't do a lot of measurements or a thorough analysis: with 10000 it seemed significantly slower, and with 65536 it seemed only a little bit faster, and with 1000000 it seemed to be perhaps significantly slower. A smaller buffer seems safer in case the performance is limited by available cache so I went for 32768 rather than 65536.